### PR TITLE
Gh-381: Migrate and update examples

### LINKED ISSUE #381: Remove gaffer-gremlin and Migrate examples

### PR DESCRIPTION
**Pull request by tb06904**
Moves the examples to a common area and updates them for new Gremlin API. Few tweaks to readmes and documentation etc too.

# Related issue

- Resolve #381

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,29 @@
 # Gaffer Docker
 
 This repo contains the code needed to run Gaffer using Docker or Kubernetes.
-There are two main sub-folders, 'docker' and 'kubernetes' which contain the project files you need for starting Gaffer using those services.
+There are two main sub-folders, 'docker' and 'kubernetes' which contain the
```
Comment (cn337131) [inline README.md:4]:
```suggestion
There are two main sub-folders, 'docker' and 'kubernetes', which contain the
```

File: README.md [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
+project files you need for starting Gaffer using those services.
 
-# Running Gaffer Using Docker
+## Running Gaffer Using Docker
 
-For information on how to run Gaffer using Docker containers, please see the documentation: [Gaffer Docker Docs](https://gchq.github.io/gaffer-doc/latest/dev/docker/)
+For information on how to run Gaffer using Docker containers, please see the
```
Comment (cn337131) [inline README.md:9]:
```suggestion
For information on how to run Gaffer using Docker containers please see the
```

File: README.md [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
 
-# Versioning
+For information on how to run Gaffer using Kubernetes, please see the
+documentation: [Gaffer Kubernetes Docs](https://gchq.github.io/gaffer-doc/latest/administration-guide/gaffer-deployment/kubernetes-guide/running-on-kubernetes.html)
 
-Each of our images which is released will be tagged with the version of the software they represent. Every release,
-we update the `latest` tag for each image and add a new release which has the corresponding version tag.
+## Versioning
+
+Each of our images which is released will be tagged with the version of the
```
Comment (cn337131) [inline README.md:24]:
```suggestion
Each of released images will be tagged with the version of the
```

File: README.md [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
+software they represent. Every release, we update the `latest` tag for each
```
Comment (cn337131) [inline README.md:25]:
```suggestion
software they represent. Every release we update the `latest` tag for each
```

File: docker/example-deployments/modern-example/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Gaffer Modern Example
+
+This is an example deployment of Gaffer on Accumulo. This uses a REST API
+deployment of Gaffer with full Accumulo and Hadoop storage backing.
+
+The example uses the
```
Comment (cn337131) [inline docker/example-deployments/modern-example/README.md:6]:
```suggestion
This example uses the
```

File: docker/example-deployments/proxy-example/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Proxy Example
+
+## Running
+
+This is an example deployment of Gaffer using a Proxy store. This uses two REST
+APIs on a Map store with data in and another as a proxy pointing to the other.
```
Comment (cn337131) [inline docker/example-deployments/proxy-example/README.md:6]: This uses two REST APIs, one contains the data and is backed by a Map Store and the other is a Proxy Store configured to forward queries from a second REST API to this Map Store backed REST.

File: README.md (hunk 3 above)
Comment (p29876) [reply README.md:24]: Each of **the/our** released images?

File: README.md [hunk 7]
```diff
@@ -24,17 +33,20 @@ If we release Gaffer version 2.1.2, the following images would be uploaded:
...
-We maintain mutable versions of latest, as well as the major, minor and bugfix versions of Gaffer. For reproducibility
-make sure to use the full version in your build metadata. For `gaffer`/`gaffer-rest` images, we also create a tag including the
-accumulo version, this allows for compatibility with Accumulo 1.9.3 in our tests. The `-accumulo-1.9.3` tagged images
-are not published but can be build locally if required.
+We maintain mutable versions of latest, as well as the major, minor and bugfix
+versions of Gaffer. For reproducibility make sure to use the full version in
+your build metadata. For `gaffer`/`gaffer-rest` images, we also create a tag
+including the accumulo version, this allows for compatibility with Accumulo
+1.9.3 in our tests. The `-accumulo-1.9.3` tagged images are not published but
+can be build locally if required.
```
Comment (p29876) [inline README.md:41]:
```suggestion
can be built locally if required.
```

File: docker/example-deployments/README.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+primarily feature the use of the Gremlin interface for interacting with the
+Graph.
+
+All examples will provide a Gaffer REST API to start interacting with the
+deployed graph.
+
+## Modern Example (Accumulo)
+
+The modern example can be found under the `modern-example` directory which
+features an Accumulo store as its storage backing.
```
[the remark this answers is not in the document]
Comment (tb06904) [reply docker/example-deployments/README.md:14]: Good point probably not overly obvious at first glance I'll add something in

**Review by cn337131 — APPROVED** [review]

**Pull request merged**